src/pdf_fmt_plug.c

```diff
@@ -318,7 +318,7 @@ static void *get_salt(char *ciphertext)
 	p = strtokm(NULL, "*");
 	cs.length = atoi(p);
 	p = strtokm(NULL, "*");
-	cs.P = atoi(p);
+	cs.P = (int)strtoll(p, (char **)NULL, 10);
```

Comment:
`atoi(p)` is equivalent to `(int)strtol(p, (char **)NULL, 10)` which is why this fix wasn't needed on systems where long is 64-bit.